include/sample_rate_converter.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ * @brief Sample Rate Converter library header.
+ */
+
+#ifndef _SAMPLE_RATE_CONVERTER_H_
+#define _SAMPLE_RATE_CONVERTER_H_
+
+/**
+ * @defgroup sample_rate_converter Sample Rate Converter library
+ * @brief Implements functionality to increase or decrease the sample rate of a data array using
+ * the CMSIS DSP filtering library.
+ *
+ * @{
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <dsp/filtering_functions.h>
+
+/**
+ * Maximum size for the internal state buffers.
+ *
+ * The internal state buffer must for each context fulfill the following equations:
+ *	Interpolation:
+ *		number of filter taps + block size - 1
+ *	Decimation:
+ *		(number of filter taps / conversion ratio) + block size - 1
+ *
+ * The equation for interpolation is used as size as this gives the largest number.
+ */
+#define SAMPLE_RATE_CONVERTER_STATE_BUFFER_SIZE                                                    \
+	CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE + CONFIG_SAMPLE_RATE_CONVERTER_MAX_FILTER_SIZE - 1
+
+/** Filter types supported by the sample rate converter */
+enum sample_rate_converter_filter {
+	SAMPLE_RATE_FILTER_SIMPLE,
```

Comment:
```suggestion
	SAMPLE_RATE_FILTER_SIMPLE  = 1,
```

Reply:
Suggest renaming SIMPLE to TRIVIAL

Reply:
I don't think I agree. Currently the simple filter still does some filtering with a very small filter, but it's still doing DSP filtering. In my head trivial is more like the implementation that was made earlier that blindly adds/removes samples, which is more of a trivial method of doing it.

Writing this comment made me realize that maybe SIMPLE and SMALL is the wrong way to go though. We could just call it SMALL and BIG, and refactor what is the big and small filters in the future _if_ we add more filters?